Verify AnkiDroid works with JDK16 for development

Currently JDK16 is the most up to date JDK version available (https://adoptopenjdk.net/archive.html?variant=openjdk16&jvmVariant=hotspot)

AnkiDroid has a test in build.gradle to make sure only JDKs we have verified are used (there have been problems in the past...) So JDK16 is not currently allowed, the most recent allowed is JDK14

The task here is to:

-  install JDK16 (adoptopenjdk.net is highly recommended - and their builds are typically available via a package manager on your machine, like brew for macos, apt for ubuntu, possibly choco for windows, etc)

- alter the build.gradle test so it allows JDK16 then verify when you build AnkiDroid that it is *using* the JDK16 binary

- run `./gradlew clean jacocoTestReport lintRelease` to make sure everything is working

Post a PR with the change the build.gradle test so it allows JDK8 and JDK11 (the two LTS versions of the JDK) as well as 16 (the current version)

_Originally posted by @mikehardy in https://github.com/ankidroid/Anki-Android/issues/8387#issuecomment-812551824_

@mikehardy after following the above tests, I got this result on running the command:
```bash
FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':api:compileDebugJavaWithJavac'.
> java.lang.IllegalAccessError: class org.gradle.internal.compiler.java.ClassNameCollector (in unnamed module @0x40668c7d) cannot access class com.sun.tools.javac.code.Symbol$TypeSymbol (in module jdk.compiler) because module jdk.compiler does not export com.sun.tools.javac.code to unnamed module @0x40668c7d

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':lint-rules:compileJava'.
> java.lang.IllegalAccessError: class org.gradle.internal.compiler.java.ClassNameCollector (in unnamed module @0xe8b0f4d) cannot access class com.sun.tools.javac.code.Symbol$TypeSymbol (in module jdk.compiler) because module jdk.compiler does not export com.sun.tools.javac.code to unnamed module @0xe8b0f4d

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.8.3/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 1m 24s
20 actionable tasks: 6 executed, 14 from cache
```

--- a/AnkiDroid/build.gradle
+++ b/AnkiDroid/build.gradle
@@ -1,6 +1,6 @@
 plugins {
  // Gradle plugin portal 
- id 'com.github.triplet.play' version '3.5.0-agp7.0'
+ id 'com.github.triplet.play' version '3.6.0'
 }
 
 apply plugin: 'com.android.application'

--- a/AnkiDroid/build.gradle
+++ b/AnkiDroid/build.gradle
@@ -140,7 +140,7 @@ android {
 
         // We want the same version stream for all ABIs in debug but for release we can split them
         if (variant.buildType.name == 'release') {
-            variant.outputs.each { output ->
+            variant.outputs.all { output ->
 
                 // For each separate APK per architecture, set a unique version code as described here:
                 // https://developer.android.com/studio/build/configure-apk-splits.html

--- a/build.gradle
+++ b/build.gradle
@@ -35,11 +35,11 @@ allprojects {
 ext {
 
     jvmVersion = Jvm.current().javaVersion.majorVersion
-    if (jvmVersion != "11" && jvmVersion != "14") {
+    if (jvmVersion != "11" && jvmVersion != "14" && jvmVersion != "16") {
         println "\n\n\n"
         println "**************************************************************************************************************"
         println "\n\n\n"
-        println "ERROR: AnkiDroid builds with JVM 11 and 14"
+        println "ERROR: AnkiDroid builds with JVM version 11, 14, or 16."
         println "  Incompatible major version detected: '" + jvmVersion + "'"
         println "\n\n\n"
         println "**************************************************************************************************************"

